gramps/gen/filters/optimizer.py

```diff
@@ -73,11 +76,17 @@ def compute_potential_handles_for_rule(
         self,
         rule: Rule,
     ) -> Tuple[Set[PrimaryObjectHandle] | None, Set[PrimaryObjectHandle] | None]:
-        """ """
+        """
+        Find the the handles for a particular rule.
+        """
+        # Has to be of the appropriate type
         if hasattr(rule, "selected_handles"):
             return (rule.selected_handles, None)
         if hasattr(rule, "find_filter"):
             filter = rule.find_filter()
-            if filter:
+            if filter and self.is_same_namespace(filter):
                 return self.compute_potential_handles_for_filter(filter)
         return (None, None)
+
+    def is_same_namespace(self, filter):
+        return self.top_level_filter.__class__.__name__ == filter.__class__.__name__
```

Comment:
I'm not 100% sure this is the right test, but the tests pass. Could probably use some more mixed-namespace filter tests.

Reply:
@dsblank I saw you comment a few days ago and will take a look...